lib/tools.js

```diff
@@ -2329,6 +2329,65 @@ function pipeLinewise(input, output) {
     rl.on('line', line => output.write(line + os.EOL));
 }
 
+/**
+ * Find the adaper main file as full path
+ *
+ * @memberof tools
+ * @param {string} adapter - adapter name of the adapter, e.g. hm-rpc
+ * @returns {Promise<string>}
+ */
+async function getAdapterMainFile(adapter) {
```

Comment:
Personally I would call the method `resolveAdapterMainFile` because in a way this is similar to what `require.resolve` does. But `get` is okay.